core/os/threaded_array_processor.h

```diff
@@ -39,11 +39,11 @@
 
 template <class C, class U>
 struct ThreadArrayProcessData {
-	uint32_t elements;
+	uint32_t elements = 0;
 	SafeNumeric<uint32_t> index;
-	C *instance;
-	U userdata;
-	void (C::*method)(uint32_t, U);
+	C *instance = nullptr;
+	U userdata = nullptr;
+	void (C::*method)(uint32_t, U){};
```

Comment:
Not sure what this does exactly, my template-fu is non-existent :P

Reply:
I'd prefer to see this struct moved to a class with a single public constructor with all the required arguments instead of having to remember to initialize every struct member manually later on:   

```c++ ThreadArrayProcessData<C, U> data;
	data.method = p_method;
	data.instance = p_instance;
	data.userdata = p_userdata;
	data.index.set(0);
	data.elements = p_elements;
	data.process(0); //process first, let threads increment for next
```

If intention is this struct can only be useful when all fields are assigned/set, then there should just be one public constructor that takes all required parameters.  You can still expose all class properties as public, but at least make sure the struct/class can only be initialized properly with a single public constructor.   This is somewhat error prone if you add another field to the internals of a struct, as you won't get help from a compiler if you forget to update something elsewhere.  I suppose at least it's confined to this source file though.

Reply:
This looks good.  `void (C::*method)(uint32_t, U){};` initializes to a null function ptr, which should give a easy to track down crash if every used improperly.  

I think this should probably just be refactored to support a single constructor considering how it's used later on in same src file however.